controller/api/destination/server.go

```diff
@@ -96,11 +94,17 @@ func (s *server) GetProfile(dest *pb.GetDestination, stream pb.Destination_GetPr
 
 	listener := newProfileListener(stream)
 
-	proxyID := strings.Split(dest.ProxyId, ".")
 	proxyNS := ""
-	// <deployment>.deployment.<namespace>.linkerd-managed.linkerd.svc.cluster.local
-	if len(proxyID) >= 3 {
-		proxyNS = proxyID[2]
+	parts := strings.Split(dest.ProxyId, ":")
+	// ns:<namespace>
+	if len(parts) == 2 && parts[0] == "ns" {
+		proxyNS = parts[1]
+	} else {
+		parts = strings.Split(dest.ProxyId, ".")
+		// <deployment>.deployment.<namespace>.linkerd-managed.linkerd.svc.cluster.local
```

Comment:
would this happen if there's an older proxy / injected deployment? if so, should we log it?

Reply:
It's really only during upgrade that this is necessary -- i.e. when an older proxy is talking to a newer control plane. If we're worried more generally about older proxies staying around, there are probably better ways to flag this than in the destination service logs.  That said, will add a comment here to remove this in the next release.